ajax/webhook.php

```diff
@@ -153,25 +157,23 @@
                 ]);
             }
         } else {
-            http_response_code(404);
-            die();
+            throw new NotFoundHttpException();
         }
         die();
     case 'resend':
         $result = QueuedWebhook::sendById($_POST['id']);
-        if ($result) {
-            http_response_code(200);
-        } else {
-            http_response_code(400);
+        if (!$result) {
+            throw new BadRequestHttpException();
         }
         die();
     case 'get_monaco_suggestions':
         header("Content-Type: application/json; charset=UTF-8");
         try {
             echo json_encode(Webhook::getMonacoSuggestions($_GET['itemtype']), JSON_THROW_ON_ERROR);
         } catch (JsonException $e) {
-            http_response_code(500);
+            throw new HttpException(500, 'Error encoding JSON.', $e);
         }
```

Comment:
You can remove the `try/catch` block here.

Reply:
> You can remove the `try/catch` block here.

Well, `JsonException` is not an HttpException, so even though it will be converted to an HTTP 500, the fact that we can add a message like "Error encoding Json" is helpful for the upcoming error handling and new error pages as proposed in #17703.

In general, having more meaningful errors is a huge helper in debugging and fixing issues, sometimes before even the customers know it :)

Reply:
Unless we provide a message that is meaningful for the end user (i.e. translated and corresponding to a functional" message), I think we should not set a specific message when throwing exception to help us identifying whether we should display the exception specific message of a generic message:

```php
$message = $e->getMessage();
if ($message === '') {
    // ... define the fallback message
}
```

Also, probably all the exceptions thrown in this PR are corresponding to some kind of security (to block handmade requests) or corresponding to a bug (and giving a feedback to the user would not help at all).